### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

**Review by sioldham — DISMISSED** [review]

**Review by bjpirt — DISMISSED** [review]

File: .github/codeql/paths.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+paths-ignore:
+  - cypress
+  - test
+  - testFixtures
+  - src/lib/config.ts
+  - src/pages/login/reset-password.tsx
+  - src/useCases/*
```
Comment (daviesjamie) [inline .github/codeql/paths.yml:7]: Should we really be ignoring all of the `useCases`? I'd say this is actually where the bulk of the code for the service is?

Comment (daviesjamie) [reply .github/codeql/paths.yml:7]: Ok, on further inspection, it turns out my intuition was wrong; the useCases only actually represent about 30% of the files and 9% of the code base:

```
❯ cloc src
     172 text files.
     172 unique files.
       1 file ignored.

github.com/AlDanial/cloc v 1.92  T=0.09 s (1839.2 files/s, 193256.9 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
TypeScript                     169            913             48          16186
JSON                             2              0              0            904
SCSS                             1              3              0             19
-------------------------------------------------------------------------------
SUM:                           172            916             48          17109
-------------------------------------------------------------------------------
```

```
❯ cloc src/useCases
      52 text files.
      52 unique files.
       0 files ignored.

github.com/AlDanial/cloc v 1.92  T=0.03 s (1864.1 files/s, 67000.7 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
TypeScript                      52            270              5           1594
-------------------------------------------------------------------------------
SUM:                            52            270              5           1594
-------------------------------------------------------------------------------
```

...But my question still stands; why are we excluding these?

Comment (bjpirt) [reply .github/codeql/paths.yml:7]: Agreed

**Review by daviesjamie — APPROVED** [review]

File: .github/codeql/paths.yml (hunk 1 above)
Comment (brettminnie) [reply .github/codeql/paths.yml]: @daviesjamie this was my assumption that UseCases where tests, hence I asked you guys to go through these ;)

**Pull request merged**